geo_classifier/train.py

```diff
@@ -0,0 +1,176 @@
+            for param in model.parameters():
+                param.requires_grad = False
+
+        model.fc = torch.nn.Linear(model.fc.in_features, self.hparams.countries_num)
+
+        return model
+
+    def forward(self, x):
+        return self.model(x)
+
+    def configure_optimizers(self):
+        optim_feature_extrator = torch.optim.SGD(
+            self.parameters(), **self.hparams.optim["params"]
+        )
+
+        return {
+            "optimizer": optim_feature_extrator,
+            "lr_scheduler": {
+                "scheduler": torch.optim.lr_scheduler.MultiStepLR(
+                    optim_feature_extrator, **self.hparams.scheduler["params"]
+                ),
+                "interval": "epoch",
+                "name": "lr",
+            },
+        }
+
+    def training_step(self, batch, batch_idx, optimizer_idx=None):
+        images, targets, true_lats, true_lngs = batch
+
+        preds = self.model(images)
+
+        loss = self.loss_fn(preds, targets)
+        self.train_acc(torch.argmax(preds, dim=1), targets)
+
+        self.log("train_loss", loss, on_step=True, on_epoch=True, prog_bar=True, logger=True)
+        self.log("train_acc", self.train_acc, on_step=True, on_epoch=True, prog_bar=True, logger=True)
+
+        return loss
+
+    def validation_step(self, batch, batch_idx):
+        images, targets, true_lats, true_lngs = batch
+
+        preds = self.model(images)
+
+        loss = self.loss_fn(preds, targets)
+        self.val_acc(torch.argmax(preds, dim=1), targets)
+
+        self.log("val_loss", loss, on_step=True, on_epoch=True, prog_bar=True, logger=True)
+        self.log("val_acc", self.val_acc, on_step=True, on_epoch=True, prog_bar=True, logger=True)
+
+    def train_dataloader(self):
+        transforms = torchvision.transforms.Compose(
+            [
+                torchvision.transforms.RandomHorizontalFlip(),
+                torchvision.transforms.RandomResizedCrop(224, scale=(0.66, 1.0)),
+                torchvision.transforms.ToTensor(),
+                torchvision.transforms.Normalize(
+                    (0.485, 0.456, 0.406), (0.229, 0.224, 0.225)
+                ),
+            ]
+        )
+
+        dataset = PhotoCoordsCountryDataset(
+            path="data/",
+            transforms=transforms,
+        )
+
+        dataloader = torch.utils.data.DataLoader(
+            dataset,
+            shuffle=True,
+            batch_size=self.hparams.batch_size,
+            num_workers=self.hparams.num_workers_per_loader,
+            pin_memory=True,
+        )
+
+        return dataloader
+
+    def val_dataloader(self):
+        transforms = torchvision.transforms.Compose(
+            [
+                torchvision.transforms.Resize(256),
+                torchvision.transforms.CenterCrop(224),
+                torchvision.transforms.ToTensor(),
+                torchvision.transforms.Normalize(
+                    (0.485, 0.456, 0.406), (0.229, 0.224, 0.225)
```

Comment:
https://stackoverflow.com/questions/58151507/why-pytorch-officially-use-mean-0-485-0-456-0-406-and-std-0-229-0-224-0-2